### PR TITLE
KAS-4598: now updates agendaitem number when changing agendaitemtype

### PR DESCRIPTION
**Pull request by StuyckRuben**
https://kanselarij.atlassian.net/browse/KAS-4598

### CODE REVIEW

File: app/components/subcase/description-panel/edit.js [hunk 1]
```diff
@@ -122,13 +122,31 @@ export default class SubcaseDescriptionEdit extends Component {
...
+    const agendaItem = await this.store.queryOne('agendaitem', {
+      'filter[agenda-activity][subcase][:id:]': this.args.subcase.id,
+      'filter[:has-no:next-version]': 't',
+    })
+
+    const latestAgendaitemOfType = await this.store.queryOne('agendaitem', {
+      'filter[agenda][agendaitems][agenda-activity][subcase][:id:]': this.args.subcase.id,
+      'filter[:has-no:next-version]': 't',
+      'filter[agenda-activity][subcase][agenda-item-type][:uri:]': agendaitemType.uri,
+      sort: 'number'
```
Comment (sergiofenoll) [inline app/components/subcase/description-panel/edit.js]: Needs to be `-number`. Sorting is ascending by default (so 1, 2, 3, ... will be returned). We want the largest number first, so we want to make it descending, using `-`.

See: https://github.com/mu-semtech/mu-cl-resources?tab=readme-ov-file#sorting

File: app/components/subcase/description-panel/edit.js [hunk 2]
```diff
@@ -122,13 +122,31 @@ export default class SubcaseDescriptionEdit extends Component {
...
+    const agendaitemType = await this.args.subcase.agendaItemType;
+    const agendaItem = await this.store.queryOne('agendaitem', {
+      'filter[agenda-activity][subcase][:id:]': this.args.subcase.id,
+      'filter[:has-no:next-version]': 't',
+    })
+
+    const latestAgendaitemOfType = await this.store.queryOne('agendaitem', {
+      'filter[agenda][agendaitems][agenda-activity][subcase][:id:]': this.args.subcase.id,
+      'filter[:has-no:next-version]': 't',
+      'filter[agenda-activity][subcase][agenda-item-type][:uri:]': agendaitemType.uri,
```
Comment (sergiofenoll) [inline app/components/subcase/description-panel/edit.js]: `'filter[type][:uri:]': agendaitemType.uri` also works and is less heavy on the query.

File: app/components/subcase/description-panel/edit.js [hunk 3]
```diff
@@ -122,13 +122,31 @@ export default class SubcaseDescriptionEdit extends Component {
     if (this.agendaItemType.uri !== oldAgendaItemType.uri) {
       await this.updateNewsletterAfterRemarkChange();
       await this.updateDecisionReports();
+      await this.recalculateAgendaitemNumber();
```
Comment (sergiofenoll) [inline app/components/subcase/description-panel/edit.js]: Recalculating after the agendaitem has been saved won't always work.

For example: this subcase is a nota with number 7. There are 3 mededelingen on the agenda. When switching the type to mededeling, the query in this function (after fixing it) will find this subcase (as 7 is larger than 3) and set its number to 8.

We need to perform this calculation before saving, or at the very least filter out the current agendaitem when trying to find the number.

**Review by sergiofenoll — CHANGES_REQUESTED** [review]

**Comment by ValenberghsSven** [comment]
When changing type, we change the decision name ('punt'; 'mededeling') but not the number.
Going to try and fix this quick

**Review by ValenberghsSven — APPROVED** [review]
New number is now also used to generate the new piece name.
Tested locally.
Can merge if jenkins is green

**Comment by ValenberghsSven** [comment]
only flaky tests are also on dev

**Pull request merged**